<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>linear_search</class>
 <widget class="QMainWindow" name="linear_search">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="lblTitle">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>801</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
    font-family: &quot;Times New Roman&quot;, &quot;Segoe UI&quot;, Arial, sans-serif;
    font-size: 28px;
    font-weight: bold;
	color: qlineargradient(
        spread:pad, 
        x1:0, y1:0, x2:1, y2:0, 
        stop:0 #0077b6, stop:1 #48cae4
    );
    qproperty-alignment: AlignCenter;
    letter-spacing: 1px;
    text-transform: uppercase;
    background: transparent;
    padding: 5px;
}

</string>
    </property>
    <property name="text">
     <string>Linear Search Algorithm Visualization</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>40</y>
      <width>791</width>
      <height>42</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="btnResetlin">
       <property name="styleSheet">
        <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnRandomizelin">
       <property name="styleSheet">
        <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
       </property>
       <property name="text">
        <string>Randomize</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnPauselin">
       <property name="styleSheet">
        <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
       </property>
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnsearchlin">
       <property name="styleSheet">
        <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
       </property>
       <property name="text">
        <string>Search</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>6</x>
      <y>520</y>
      <width>791</width>
      <height>43</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QSlider" name="sliderSpeed">
       <property name="minimumSize">
        <size>
         <width>211</width>
         <height>41</height>
        </size>
       </property>
       <property name="minimum">
        <number>100</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>500</number>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TickPosition::TicksBelow</enum>
       </property>
       <property name="tickInterval">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="lblOperation">
       <property name="styleSheet">
        <string notr="true">QLabel {
    font-family: &quot;Times New Roman&quot;, &quot;Segoe UI&quot;, Arial, sans-serif;
    font-size: 14px;
    font-weight: bold;
	color: qlineargradient(
        spread:pad, 
        x1:0, y1:0, x2:1, y2:0, 
        stop:0 #0077b6, stop:1 #48cae4
    );
    qproperty-alignment: AlignCenter;
    letter-spacing: 1px;
    text-transform: uppercase;
    background: transparent;
    padding: 5px;
}

</string>
       </property>
       <property name="text">
        <string>TextLabel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnBacklin">
       <property name="styleSheet">
        <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 12px;
    color: white;
	background: qlineargradient(
    	spread:pad,
    	x1:0, y1:0, x2:1, y2:1,
    	stop:0 #000000,
    	stop:1 #1a1a1a
	);

    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}


</string>
       </property>
       <property name="text">
        <string>back</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>781</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
    font-family: &quot;Times New Roman&quot;, &quot;Segoe UI&quot;, Arial, sans-serif;
    font-size: 7px;
    font-weight: bold;
	color: qlineargradient(
        spread:pad, 
        x1:0, y1:0, x2:1, y2:0, 
        stop:0 #ffffff
    );
    qproperty-alignment: AlignCenter;
    letter-spacing: 1px;
    text-transform: uppercase;
    background: transparent;
    padding: 5px;
}

</string>
    </property>
    <property name="text">
     <string>Enter the number you want to search :</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineSearchEdit">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>100</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
